astra/src/test/java/com/slack/astra/logstore/opensearch/OpenSearchAdapterTest.java

```diff
@@ -78,6 +80,7 @@ public OpenSearchAdapterTest() throws IOException {
             true,
             true));
     openSearchAdapter = new OpenSearchAdapter(fieldDefBuilder.build());
+    System.setProperty("astra.query.useOpenSearchParsing", "true");
```

Comment:
This is dangerous, as it sets this for the rest of the tests and not just this one. An easy way to handle this is to read the value from the sysprop at the test init, and then in the cleanup reset it back to what it was or delete it if not originally provided. Since this variable is only expected to have a short lifespan that might be ideal.

Also consider switching the default value to inverted - so all tests run with it as true, and then override it to false at deploy time.